app/verify_workflow.go

```diff
@@ -177,9 +182,18 @@ func getStepName(step string) string {
 }
 
 func getStepUses(step *actionlint.Step) *actionlint.String {
+	// TODO("naveen") :- Remove this check as this never happens
+	// errors during actionlint.Parse: [:16:3: "steps" section is
+	// missing in job "analysis" [syntax-check]]
+	// workflow-invalid-nil-steps.yml has nil steps
 	if step == nil || step.Exec == nil {
 		return nil
 	}
+	// TODO("naveen") :- Remove this check
+	// errors during actionlint.Parse: [:29:9: this step is for running action since it contains
+	// at least one of "uses", "with" keys, but also contains "run" key which is used for running
+	// shell command [syntax-check]]
+	// workflow-invalid-execaction.yml has execaction
```

Comment:
I think this check is a Golang best practice so would leave it as is. We are doing type check assertion so we are just confirming that the assertion was true.

Reply:
I agree. I have removed the comment.